src/forms/Textarea.jsx

```diff
@@ -8,29 +8,61 @@ import autosize from 'autosize';
 class Textarea extends React.Component {
 	constructor(props) {
 		super(props);
+		this.onChange = this.onChange.bind(this);
+
 		this.state = {
-			value: props.value || '',
+			value: '',
 		};
-		this.onChange = this.onChange.bind(this);
 	}
 
+	/**
+	 * Turns on autosize if requested
+	 * @return {undefined} side effect only
+	 */
+	componentDidMount() {
+		if (this.props.rows === 'auto'){
+			autosize(this.textarea);
+		}
+	}
+
+	/**
+	 * @param {Object} props the incoming props
+	 * @return {undefined} side effect only
+	 */
+	componentWillMount() {
+		this.setState((state, props) => ({
+			value: props.value || ''
+		}));
+	}
+
+	/**
+	 * @param {Object} nextProps the incoming props
+	 * @return {undefined} side effect only
+	 */
+	componentWillReceiveProps(nextProps) {
+		this.setState((state, props) => ({
+			value: nextProps.value || ''
+		}));
```

Comment:
since it's the same as `componentWillMount`, let's make this a new method that both lifecycle methods can call. A name like `overrideValue` will make the intent more clear as well